### PR TITLE
feat: add cta to add liqudity for fox farming

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

- Add a cta to go add liquidity for FOX farming.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #

### Operations

When clicking on fox farming and you have no balance of FOX/ETH LP you will be given a CTA to provide liquidity.

If you do have a balance of LP tokens you will see continue.

## Screenshots (if applicable)

Screen Shot 2023-04-03 at 6 11 24 PM

### CODE REVIEW

File: src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx [hunk 1]
```diff
@@ -1,33 +1,76 @@
...
+    useAppSelector(state => selectPortfolioFiatBalanceByAssetId(state, { assetId })) ?? '0'
+  const history = useHistory()
+  const {
+    receive: { open },
+  } = useModal()
+
+  const handleLpDepositClick = useCallback(() => {
+    return history.push(
+      '?type=lp&provider=Uniswap%20V2&chainId=eip155%3A1&assetNamespace=erc20&assetReference=0x470e8de2ebaef52014a47cb5e6af86884947f08c&rewardId=&modal=deposit',
+    )
```
Comment (0xApotheosis) [inline src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx:43]: We might want to extract this string to a constant.

**Review by 0xApotheosis — COMMENTED** [review]
Does what it says on the box, though a few thoughts:

1. Having a modal on top of a modal when clicking "receive" is a bit rough
2. When clicking "Provide Liquidity", and then going back, you end up in a different (and new) place, which is a little disconcerting

File: src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx:43]: Betcha we did image

File: src/assets/translations/en/main.json [hunk 2]
```diff
@@ -278,13 +278,15 @@
       },
       "foxFarmingOverview": {
         "header": "Deposit ETH-FOX LP tokens to %{opportunity} to earn",
-        "body": "Approve and stake your liquidity tokens to earn bonus FOX rewards.",
+        "body": "Approve and stake your FOX-ETH UniswapV2 liquidity pool tokens tokens to earn bonus FOX rewards.",
```
Comment (gomesalexandre) [inline src/assets/translations/en/main.json:281]: We may also want with "LP tokens" for conciseness 
```suggestion
        "body": "Approve and stake your ETH/FOX UniswapV2 liquidity pool tokens tokens to earn bonus FOX rewards.",
```

File: src/assets/translations/en/main.json [hunk 3]
```diff
@@ -278,13 +278,15 @@
...
         "rewards": "The rewards you earn will accrue automatically.",
         "cta": "Deposit ETH-FOX LP tokens Now",
         "foxFarmingBalance": "ETH-FOX LP tokens Balance",
         "withdrawals": "Withdrawals",
         "emptyWithdraws": "Pending or active withdrawals will show here.",
-        "availableDate": "available %{date}"
+        "availableDate": "available %{date}",
+        "emptyBalanceBody": "or provide liqudity for FOX & ETH on UniswapV2 to start earning.",
```
Comment (gomesalexandre) [inline src/assets/translations/en/main.json:288]: The reason is ETH and FOX are in this order in the pool, referencing it as FOX/ETH was a mistake we propagated everywhere

```suggestion
        "emptyBalanceBody": "or provide liqudity for ETH & FOX on UniswapV2 to start earning.",
```

File: src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx (hunk 1 above)
Comment (reallybeard) [reply src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx:43]: This is just the contract address, need the full URL to direct them to.

**Comment by reallybeard** [comment]
> Does what it says on the box, though a few thoughts:
> 
> 1. Having a modal on top of a modal when clicking "receive" is a bit rough
> 2. When clicking "Provide Liquidity", and then going back, you end up in a different (and new) place, which is a little disconcerting

Not sure how we can make back take them back to the other url, I guess we can pass a like return url or something that overrides the back button.

File: src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx [hunk 4]
```diff
@@ -1,33 +1,76 @@
...
 import { Amount } from 'components/Amount/Amount'
-import { Text } from 'components/Text'
+import { RawText, Text } from 'components/Text'
+import { useModal } from 'hooks/useModal/useModal'
+import { selectAssetById, selectPortfolioFiatBalanceByAssetId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
 
 type FoxFarmingEmptyProps = {
   assets: Asset[]
+  assetId: AssetId
```
Comment (gomesalexandre) [inline src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx:18]:
```suggestion
  lpAssetId: AssetId
```

File: src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/features/defi/providers/fox-farming/components/FoxFarmingManager/Overview/FoxFarmingEmpty.tsx:43]: Can also reference `lpAssetId` we pass as props

**Review by gomesalexandre — COMMENTED** [review]
In addition to @0xApotheosis spotted issue WRT the back button and UX suggestion, a few more here:

- If we keep "Receive" as a modal, we should probably disable the back button, since the intent is to receive this asset specifically
- "Receive" as a text span looks off, could we make it more CTA-y e.g as a button?

**Pull request closed without merging**